slurm/slurm.sh

```diff
@@ -0,0 +1,300 @@
+
+slurm_submit(){
+    local commandString="$1"
+    local jobQueue="$2"
+    local jobName="$3"
+    local slurmMem="$4"
+    local nThreads="$5"
+    local jobGroupName="$6"       
+    local workingDir="$7"
+    local logPrefix="$8"
+    local prioritise="$9"
+    local condaEnv="${10}"
+    local quiet="${11:-'no'}"
+
+    # Need at least the command string
+
+    if [ -z "$commandString" ]; then
+        die "Need at least a command string for SLURM submission"
+    fi
+
+    # Check parameter settings
+
+    if [ -n "$jobQueue" ]; then jobQueue=" -p ${jobQueue}"; fi
+    if [ -n "$jobName" ]; then jobName=" --J ${jobName}"; fi
+    if [ -n "$slurmMem" ]; then slurmMem=" --mem $slurmMem"; fi
+    if [ -n "$nThreads" ]; then nThreads=" --cpus-per-task $nThreads"; fi
+    if [ -n "$jobGroupName" ]; then jobGroupName=" --job-group $jobGroupName"; fi
+    if [ -n "$workingDir" ]; then workingDir=" --chdir \"$workingDir\""; fi
+    if [ -n "$condaEnv" ]; then
+        condaBase=$(conda info --json | awk '/conda_prefix/ { gsub(/"|,/, "", $2); print $2 }')
+        #condaCmd=". ${condaBase}/bin/activate ${condaBase}/envs/${condaEnv}"
+        ###### the lines below were added to enable the activation of a conda env that is not inside the base conda env directory
+        condaEnvPath="${condaBase}/envs/${condaEnv}"
+        if [ -d "$condaEnv" ]; then condaEnvPath="$condaEnv"; fi
+        condaCmd=". ${condaBase}/bin/activate ${condaEnvPath}"
+        ######
+        commandString="${condaCmd} && ${commandString}"
+    fi
+    if [ -n "$logPrefix" ]; then 
+        mkdir -p $(dirname $logPrefix)
+        logPrefix=" -o \"${logPrefix}.out\" -e \"${logPrefix}.err\""
+    fi
+    maxTime=$(slurm_maxtime_for_partition "production")
+    local sbatch_cmd=$(echo -e "sbatch -t $maxTime $jobQueue $jobName $slurmMem $nThreads $jobGroupName $workingDir $logPrefix --wrap \"$commandString\"" | tr -s " ")
+
+    local sbatchOutput=
+    sbatchOutput=$(eval $sbatch_cmd)
+
+    # Assuming submission was successful, extract the job ID
+
+    if [ $? -ne 0 ]; then
+        die "Job submission failed"
+    else
+        local jobId=$(echo $sbatchOutput | grep -oE 'Submitted batch job [0-9]+')
+        job_id=${jobId##* }
+        # if [ "$prioritise" = 'yes' ]; then
+        #     warn "Prioritising $jobId" "$quiet"
+        #     btop $jobId
+        # fi
+        echo $job_id
+    fi
+}
+
+# Get MaxTime for the partition/queue
+
+slurm_maxtime_for_partition(){
+    partition_name=$1
+    partition_info=$(scontrol show partition "$partition_name" 2>/dev/null)
+
+    if [ -z "$partition_info" ]; then
+        echo "Partition not found or permission denied."
+        return 1
+    fi
+
+    max_time=$(echo "$partition_info" | awk -F' ' '{for(i=1;i<=NF;i++) if($i ~ /^MaxTime=/) {split($i, a, "="); print a[2]}}')
+    
+    if [ -z "$max_time" ]; then
+        echo "MaxTime not found for partition $partition_name."
+        return 1
+    fi
+
+    echo "$max_time"
+}
```

Comment:
maybe this function can be moved to `generic_routines.sh`?

Reply:
but it is slurm specific

Reply:
in `generic_routines.sh` you also have LSF routines